framework/src/controller/controller.ts

```diff
@@ -126,7 +126,7 @@ export class Controller {
 				? plugins
 				: [...Object.keys(this._inMemoryPlugins), ...Object.keys(this._childProcesses)];
 
-		const errors = [];
+		let hasError = false;
```

Comment:
🤔 why is this here? it was merged in separate PR?

Reply:
without this PR, the test doesn't pass and without this fix, the deepscan won't pass, so i merged it here